src/main/java/view/DashboardView.java

```diff
@@ -81,93 +96,201 @@ public DashboardView(DashboardViewModel dashboardViewModel,
-
-                            // TODO: add reset balance pop-up
+                evt -> {
+                    resetBalanceController.execute();
+                    dashboardViewModel.firePropertyChanged();
 
-                        }
-                    }
+                    // TODO: add reset balance pop-up
                 }
         );
 
         history.addActionListener(
-                new ActionListener() {
-                    public void actionPerformed(ActionEvent evt) {
-                        if (evt.getSource().equals(history)) {
-                            historyViewModel.firePropertyChanged();
-                            viewManagerModel.setActiveView(historyViewModel.getViewName());
-                            viewManagerModel.firePropertyChanged();
-                            System.out.println("History button pressed");
-                            // TODO: Do we then need to call GetTransactionHistoryController.execute()
-                            //  to obtain the needed information?
-                        }
-                    }
+                evt -> {
+                    historyViewModel.firePropertyChanged();
+                    viewManagerModel.setActiveView(historyViewModel.getViewName());
+                    viewManagerModel.firePropertyChanged();
                 }
         );
 
-        // NOTE: may or may not be necessary
-        this.setLayout(new BoxLayout(this, BoxLayout.Y_AXIS));
-
         // Make the various display items visible to the user on the display.
-        this.add(title);
-        this.add(welcome);
-        this.add(buttons);
+        this.setLayout(new BoxLayout(this, BoxLayout.X_AXIS));
+        setLayout(new BorderLayout());
+        GridBagConstraints gbc = new GridBagConstraints();
+
+        topPanel = new JPanel(new GridBagLayout());
+        JPanel intermediatePanel1 = new JPanel(new GridBagLayout());
+        middlePanel = new JPanel(new GridBagLayout());
+        JPanel intermediatePanel2 = new JPanel(new GridBagLayout());
+        bottomPanel = new JPanel(new GridBagLayout());
+
+        gbc.gridx = 0;
+        gbc.gridy = 0;
+        gbc.anchor = GridBagConstraints.WEST;
+//        buttons.setPreferredSize(new Dimension(30, 80));
+        topPanel.add(buttons, gbc);
+
+        gbc.gridx = 1;
+        gbc.weightx = 1;
+        gbc.fill = GridBagConstraints.HORIZONTAL;
+        topPanel.add(Box.createHorizontalGlue(), gbc);
+
+        // Spice up the labels a bit.
+        Font labelFont = balanceLabel.getFont();
+//        balanceLabel.setForeground(new Color(14, 20, 138));
+        balanceLabel.setSize(25, 5);
+        balanceLabel.setFont(labelFont.deriveFont(Font.BOLD));
+
+        // Spice up the actual balance (field) as well.
+        Font fieldFont = balanceField.getFont();
+        balanceField.setForeground(Color.DARK_GRAY);
+        balanceField.setFont(fieldFont.deriveFont(Font.ITALIC));
```

Comment:
nice font usage

Reply:
Thanks! I spent a little too long figuring that out :)